.github/workflows/maven-publish.yml

```diff
@@ -86,6 +91,11 @@ jobs:
               "id": "mks-nexus-private-releases",
               "username": "${{ secrets.NEXUS_USERNAME }}",
               "password": "${{ secrets.NEXUS_PASSWORD }}"
+            },
+            {
+              "id": "docker.io",
+              "username": "${{ secrets.DOCKER_HUB_USERNAME || 'user'}}",
+              "password": "${{ secrets.DOCKER_HUB_PASSWORD || 'password'}}"
             }]
```

Comment:
What's happen if you don't provide the defaults?
So If `DOCKER_HUB_USERNAME` and `DOCKER_HUB_PASSWORD` are not set, the action will use the literal strings 'user' and 'password' as the credentials. This will result in authentication failures when trying to authenticate with Docker Hub, as these are not valid credentials.

How about:
1. Default to empty e.g `"username": "${{ secrets.DOCKER_HUB_USERNAME || ''}}"`, I guess it's the same thing, i prefer empty rather than invalid credentials.
2. Add extra step to edit `settings.xml` and add docker.io if docker credentials is required.